src/bsoncxx/test/v_noabi/bson_get_values.cpp

```diff
@@ -387,4 +390,243 @@ TEST_CASE("document::values have a superset of document::view's methods") {
     }
 }
 
+TEST_CASE("v1", "[bsoncxx][v_noabi][document][element]") {
+    using v1 = v1::element::view;
+    using v_noabi = v_noabi::document::element;
+    using bsoncxx::v_noabi::from_v1;
+    using bsoncxx::v_noabi::to_v1;
+
+    auto const owner = bsoncxx::from_json(R"({"x": 1})");
+    auto const doc = owner.view();
+    auto const e = doc["x"];
+
+    SECTION("from_v1") {
+        v1 from{e};
+        v_noabi const to = from;
+
+        CHECK(to.raw() == to_v1(to).raw());
+        CHECK(to.raw() == from.raw());
+        CHECK(to.raw() != v1{}.raw());
+    }
+
+    SECTION("to_v1") {
+        v_noabi from{e};
+        v1 const to{from};
+
+        CHECK(to.raw() == v_noabi{to}.raw()); // No from_v1().
+        CHECK(to.raw() == from.raw());
+        CHECK(to.raw() != v_noabi{}.raw());
+    }
+}
+
+TEST_CASE("v1", "[bsoncxx][v_noabi][document][view]") {
+    using v1 = v1::document::view;
+    using v_noabi = v_noabi::document::view;
+    using bsoncxx::v_noabi::from_v1;
+    using bsoncxx::v_noabi::to_v1;
+
+    auto const owner = bsoncxx::from_json(R"({"x": 1})");
+    auto const doc = owner.view();
+
+    SECTION("from_v1") {
+        v1 from{doc.data()};
+        v_noabi const to = from;
+
+        CHECK(to == to_v1(to));
+        CHECK(to == from);
+        CHECK(to != v1{});
+    }
+
+    SECTION("to_v1") {
+        v_noabi from{doc.data(), doc.size()};
+        v1 const to{from};
+
+        CHECK(to == from_v1(to));
+        CHECK(to == from);
+        CHECK(to != v_noabi{});
+    }
+}
+
+template <typename Deleter, typename T>
+bsoncxx::v1::stdx::optional<Deleter> get_deleter(T const& v) {
+    if (auto const deleter_ptr = v.get_deleter().template target<Deleter>()) {
+        return *deleter_ptr;
+    }
+    return {};
+}
+
+TEST_CASE("v1", "[bsoncxx][v_noabi][document][value]") {
```

Comment:
The refactored `v_noabi::document::value` class received the most thorough test coverage due to complexity of satisfying the deleter type invariant throughout both copy and move operations.